packages/meditrak-server/src/routes/exportSurveyResponses.js

```diff
@@ -81,71 +86,21 @@ export async function exportSurveyResponses(req, res) {
-    if (countryCode) {
-      country = await models.country.findOne({ code: countryCode });
-      countryId = country.id;
-    }
-    if (entityCode) {
-      const entity = await models.entity.findOne({ code: entityCode });
-      country = await models.country.findOne({ code: entity.country_code });
-      entities = [entity];
-    } else if (countryId) {
-      country = await models.country.findById(countryId);
-      entities = await models.entity.find({ country_code: country.code }, { sort: ['name ASC'] });
-    } else if (entityIds) {
-      entities = await Promise.all(
-        entityIds.split(',').map(entityId => models.entity.findById(entityId)),
-      );
-      if (!countryId && entities.length > 0) {
-        const countryCodeFromEntity = entities[0].country_code;
-        country = await models.country.findOne({ code: countryCodeFromEntity });
-        countryId = country.id;
-      }
-    } else if (surveyResponseId) {
-      surveyResponse = await models.surveyResponse.findById(surveyResponseId);
-      surveyId = surveyResponse.survey_id;
-      country = await surveyResponse.country();
-    } else {
-      throw new ValidationError(
-        'Please specify either surveyResponseId, countryId, countryCode, facilityCode or entityIds',
-      );
-    }
-    let surveys;
-    if (surveyId) {
-      // A surveyId of interest passed in, only export that
-      surveys = [await models.survey.findById(surveyId)];
-    } else if (surveyCodes) {
-      const surveyFindConditions = {
-        // surveyCodes may be passed through as a comma separated string or as an array, which looks like this in a query:
-        // ?surveyCodes=code1&surveyCodes=code2
-        code: {
-          comparisonType: 'whereIn',
-          args: Array.isArray(surveyCodes) ? [surveyCodes] : [surveyCodes.split(',')],
-        },
-      };
-      if (countryId) {
-        // Fetch surveys where country_ids is empty (enabled in all countries) or contains countryId
-        // eslint-disable-next-line no-underscore-dangle
-        surveyFindConditions._and_ = {
-          country_ids: '{}',
-          _or_: {
-            country_ids: { comparator: '@>', comparisonValue: [countryId] },
-          },
-        };
-      }
-      surveys = await models.survey.find(surveyFindConditions);
-    } else {
-      // No specific surveyId passed in, so export all surveys that apply to this country
-      const allSurveys = await models.survey.all();
-      surveys = allSurveys.filter(
-        survey => survey.country_ids.length === 0 || survey.country_ids.includes(countryId),
-      );
-    }
-    if (!surveys || surveys.length < 1) {
-      throw new ValidationError('Survey not found. Please check permissions');
-    }
+    const variablesExtractor = new SurveyResponseVariablesExtractor(models);
+    const reportName = viewId && (await models.dashboardReport.findById(viewId)).viewJson.name;
+    const variables = await variablesExtractor.getParametersFromInput(
```

Comment:
Nice work! I think this is a good abstraction that makes it much easier to read the code.